### ISSUE TITLE
Add optional chain argument to starknet.id hooks

### ISSUE THREAD

**Issue by rsodre** [apibara/starknet-react#420]
For a better testnet dapp experience, and to enable access to starknet.id on L3s, it is required to fetch starknet.id from mainnet instead of the connected chain.

Suggestion: Add an optional `chain` argument (SN_MAIN) top `useStarkAddress ()`, `useStarkName ()` and `useStarkProfile()`

**Comment by fracek** [comment]
Yes, happy to accept a contribution for this. The hooks should accept a new optional argument `chainId?: bigint` and use that to lookup which contracts to use.

**Comment by raizo07** [comment]
Hello @fracek I'm open to work on this.

**Comment by fracek** [comment]
Hey @raizo07 I assigned the issue to you.

**Comment by raizo07** [comment]
Alright, I'm on it.

**Comment by raizo07** [comment]
Hi @fracek based on this [documentation](https://community.starknet.io/t/wallet-switch-starknet-chain-rpc-method-wallet-switchstarknetchain/102034) `chainId` is to be a string. "A 0x-prefixed hexadecimal string".

But in the comment above you suggested we use a `bigint`. Should I proceed with a `bigint` or  a `string`

**Comment by fracek** [comment]
Hey, yes starknet-react uses a `bigint` as chainid because IMHO it works much better than hex-encoded values. To test the hook you can use a chainid from `@starknet-react/chains`, e.g. `mainnet.id`.
